core/services/job/orm.go

```diff
@@ -253,18 +254,51 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
 					if !useForwarders && err != nil {
 						return errors.Wrapf(ErrNoSuchTransmitterKey, "%v", jb.OCR2OracleSpec.TransmitterID)
 					}
+
+					newChainID, err := EVMChainIDForJobSpec(jb.OCR2OracleSpec)
+					if err != nil {
+						return err
+					}
+
+					var spec OCR2OracleSpec
+					chainIdPath := []string{"chainID"}
+					err = tx.Get(&spec, `SELECT * FROM ocr2_oracle_specs WHERE relay = $1 AND contract_id = $2 AND relay_config #>> $3 = $4 LIMIT 1`,
+						"EVM", jb.OCR2OracleSpec.ContractID, chainIdPath, fmt.Sprintf("%v", newChainID),
+					)
+
+					if !errors.Is(err, sql.ErrNoRows) {
+						if err != nil {
+							return errors.Wrapf(err, "db read error while validating contract_id")
+						}
+						chainID, err := EVMChainIDForJobSpec(&spec)
+						if err != nil {
+							return errors.Wrapf(err, "internal error while validating contract_id")
+						}
+						if chainID == newChainID {
+							return errors.Errorf("a job with contract address %v already exists for chain ID %v", jb.OCROracleSpec.ContractAddress, chainID)
+						}
```

Comment:
Is this redundant, since the query already matched?
Would it be helpful to the user to include the conflicting job id?